gramps/gen/filters/rules/person/_hascommonancestorwithfiltermatch.py

```diff
@@ -66,23 +66,13 @@ def prepare(self, db, user):
         self.with_people = []
         self.filt = MatchesFilter(self.list)
         self.filt.requestprepare(db, user)
-        if user:
-            user.begin_progress(
-                self.category,
-                _("Retrieving all sub-filter matches"),
-                db.get_number_of_people(),
-            )
-        for person in db.iter_people():
-            if user:
-                user.step_progress()
-            if person and self.filt.apply_to_one(db, person):
-                # store all people in the filter so as to compare later
-                self.with_people.append(person.handle)
-                # fill list of ancestor of person if not present yet
-                if person.handle not in self.ancestor_cache:
-                    self.add_ancs(db, person)
-        if user:
-            user.end_progress()
+
+        for handle in self.filt.find_filter().apply(db):
```

Comment:
Is this loop now fast enough that user progress updates are no longer justified\required?

Reply:
This comment also applies to the other loops where `user.step_progress()` is no longer called

Reply:
It isn't that the loop is fast enough to no longer need to update the user, but the expensive part has been moved to "apply()". If we need to update the user, it should happen there (and I think it already does).